src/adm/filter.rs

```diff
@@ -234,26 +236,23 @@ impl AdmFilter {
             path.to_mut().push('/');
         }
 
-        for filter in &filter.advertiser_urls {
-            if !host.eq(&filter.host) {
-                continue;
-            }
-
-            if let Some(ref paths) = filter.paths {
+        for filter in filters {
+            if host == filter.host {
+                let paths = &filter
+                    .paths
+                    .clone()
+                    .unwrap_or_else(|| [PathFilter::default()].to_vec());
```

Comment:
Nit: you could avoid this clone here by keeping a static of the default:

```suggestion
                    .as_ref()
                    .unwrap_or_else(|| &DEFAULT_PATH_FILTER);
```

with this in the `lazy_static` block:
```rust
static ref DEFAULT_PATH_FILTER: Vec<PathFilter> = vec![PathFilter::default()];
```